keydata_legacy_test.go

```diff
@@ -0,0 +1,334 @@
+}
+
+func (h *mockPlatformKeyDataHandler) RecoverKeysWithAuthKey(data *PlatformKeyData, encryptedPayload []byte, key []byte) ([]byte, error) {
+	if !h.passphraseSupport {
+		return nil, errors.New("not supported")
+	}
+
+	if err := h.checkState(); err != nil {
+		return nil, err
+	}
+
+	handle, err := h.unmarshalHandle(data)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := h.checkKey(handle, key); err != nil {
+		return nil, err
+	}
+
+	return h.recoverKeys(handle, encryptedPayload)
+}
+
+func (h *mockPlatformKeyDataHandler) ChangeAuthKey(data *PlatformKeyData, old, new []byte) ([]byte, error) {
+	if !h.passphraseSupport {
+		return nil, errors.New("not supported")
+	}
+
+	if err := h.checkState(); err != nil {
+		return nil, err
+	}
+
+	handle, err := h.unmarshalHandle(data)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := h.checkKey(handle, old); err != nil {
+		return nil, err
+	}
+
+	m := hmac.New(func() hash.Hash { return crypto.SHA256.New() }, handle.Key)
+	m.Write(new)
+	handle.AuthKeyHMAC = m.Sum(nil)
+
+	return json.Marshal(&handle)
+}
+
+type mockKeyDataReader struct {
+	readableName string
+	io.Reader
+}
+
+func (r *mockKeyDataReader) ReadableName() string {
+	return r.readableName
+}
+
+type keyDataTestBase struct {
+	handler *mockPlatformKeyDataHandler
+}
+
+func (s *keyDataTestBase) SetUpSuite(c *C) {
+	s.handler = &mockPlatformKeyDataHandler{}
+	RegisterPlatformKeyDataHandler(mockPlatformName, s.handler)
+}
+
+func (s *keyDataTestBase) SetUpTest(c *C) {
+	s.handler.state = mockPlatformDeviceStateOK
+	s.handler.passphraseSupport = false
+}
+
+func (s *keyDataTestBase) TearDownSuite(c *C) {
+	RegisterPlatformKeyDataHandler(mockPlatformName, nil)
+}
+
+func (s *keyDataTestBase) newKeyDataKeys(c *C, sz1, sz2 int) (DiskUnlockKey, PrimaryKey) {
+	key := make([]byte, sz1)
+	auxKey := make([]byte, sz2)
+	_, err := rand.Read(key)
+	c.Assert(err, IsNil)
+	_, err = rand.Read(auxKey)
+	c.Assert(err, IsNil)
+	return key, auxKey
+}
+
+func (s *keyDataTestBase) mockProtectKeys(c *C, key DiskUnlockKey, auxKey PrimaryKey, modelAuthHash crypto.Hash) (out *KeyParams) {
+	payload := MarshalKeys(key, auxKey)
+
+	k := make([]byte, 48)
+	_, err := rand.Read(k)
+	c.Assert(err, IsNil)
+
+	handle := mockPlatformKeyDataHandle{
+		Key: k[:32],
+		IV:  k[32:]}
+
+	h := hmac.New(func() hash.Hash { return crypto.SHA256.New() }, handle.Key)
+	handle.AuthKeyHMAC = h.Sum(nil)
+
+	b, err := aes.NewCipher(handle.Key)
+	c.Assert(err, IsNil)
+	stream := cipher.NewCFBEncrypter(b, handle.IV)
+
+	out = &KeyParams{
+		PlatformName:      mockPlatformName,
+		Handle:            &handle,
+		EncryptedPayload:  make([]byte, len(payload)),
+		PrimaryKey:        auxKey,
+		SnapModelAuthHash: modelAuthHash}
+	stream.XORKeyStream(out.EncryptedPayload, payload)
+	return
+}
+
+type keyDataLegacySuite struct {
+	keyDataTestBase
+}
+
+func (s *keyDataLegacySuite) newKeyDataKeys(c *C, sz1, sz2 int) (DiskUnlockKey, PrimaryKey) {
+	key := make([]byte, sz1)
```

Comment:
It looks like there's 2 implementations of this.

Reply:
Removed the redundant one from `keyDataLegacySuite`.